apps/full-site-editing/full-site-editing-plugin/full-site-editing/lib/site-options/with-site-options.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { createHigherOrderComponent, pure } from '@wordpress/compose';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useSiteOptions } from './use-site-options';
+
+/**
+ * Higher Order Component used to inject site options.
+ *
+ * The option value is requested from the WordPress API. Note that arbitrary
+ * options can only be retrieved if they have been registered as site settings.
+ *
+ * Updates are persisted to the database when the post is saved.
+ *
+ * In the component, every specified `optionName` will be a prop which maps to
+ * an object with the keys `value`, `updateValue`, and `loaded`. `value`
+ * contains the actual value of the WordPress option, and `updateValue` can be
+ * used to update the option. `updateValue` will re-render the component with
+ * the new `value`.
+ *
+ * @param {Object} options An object of site options to retrieve. Keys should be
+ *                         option names you wish to have locally. They can be
+ *                         named whatever makes sense in the context of your
+ *                         component. Any key's value should be another object
+ *                         whih contains `optionName`, the actual name of the
+ *                         WordPress option to fetch, and `defaultValue`, the
+ *                         value to use while the option is fetching.
+ *
+ * @return {Component} The higher order component.
+ *
+ * @example
+ * ```js
+ * function Component( { mySiteOption } ) {
+ *   const { value, updateValue } = mySiteOption;
+ *   // `updateValue( 'foo' )` the component will re-render if called.
+ *   return <span>{ value }</span>;
+ * }
+ *
+ * export default compose( [
+ *   withSiteOptions( {
+ *     mySiteOption: { optionName: 'title', defaultValue: __( 'Site title loading…' ) },
+ * 	} ),
+ * ] )( Component );
+ * ```
+ * In the above example, we call withSiteOptions with an object mapping our name
+ * for the option to some information about it. In particular, optionName needs
+ * to match the name of the option in WordPress, and defaultValue is used while
+ * the option is loading from the API.
+ *
+ */
```

Comment:
I really love how detailed this docblock is, but I think it can be more syntactically correct!

It's not markdown, so the `@example` tag doesn't need the backticks to automagically highlight (if possible) the code following it.
Unfortunately, as far as I know, there is no "close example" marker (you can stop the example by having another JSDoc tag after). We can work around this issue by moving the example comment before the code. E.g.
```
/*
 * In the following example...
 *
 * @example
 * function Component( { mySiteOption } ) {}
 */
```

Reply:
Ah interesting! I was trying to follow a similar example in Gutenberg. Will try it without the back ticks.